homeassistant/components/browan/models.py

```diff
@@ -0,0 +1,36 @@
+"""Defines models used by the integration."""
+from homeassistant.components.sensor import SensorDeviceClass
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL,
+    ATTR_TEMPERATURE,
+    UnitOfElectricPotential,
+    UnitOfTemperature,
+)
+
+
+class SensorTypes:
+    """Collection of sensor entities defining measurements of a device."""
+
+    class SensorType:
+        """Base entity type."""
+
+        DATA_KEY: str
+        UNIT: str
+        DEVICE_CLASS: SensorDeviceClass
+        NAME: str
+
+    class BatteryLevel(SensorType):
```

Comment:
There is already a solution for this available in Home Assistant. There are dataclasses for `EntityDescriptions`, in the sensor case `SensorEntityDescription`. Take a look at [Unifi Integration](https://github.com/home-assistant/core/blob/dev/homeassistant/components/unifi/sensor.py#L133)

Reply:
Thanks, updated in 681b6e1886